src/components/SearchResultList/FilterForm.tsx

```diff
@@ -0,0 +1,118 @@
+'use client';
+
+import * as React from 'react';
+import Select from 'react-select';
+
+import { Formik } from 'formik';
+
+import { SearchTypeOptions, SearchScopeOptions } from '@/constants/constants';
+import { SearchScope, SearchType } from '@/constants/enum';
+import { SearchResultFilter } from '@/types/query';
+
+interface InjectedProps {
+  onApplyFilter: (filterParams: SearchResultFilter) => void;
+  onResetFilter: () => void;
+}
+
+interface FormValues {
+  searchType: { value: SearchType; label: string } | null;
+  searchScope: { value: SearchScope; label: string } | null;
+  queryInput: string;
+}
+
+const FilterForm = (props: InjectedProps) => {
+  const { onApplyFilter, onResetFilter } = props;
+
+  const initialValues: FormValues = {
+    searchType: null,
+    searchScope: null,
+    queryInput: '',
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onReset={() => {
+        console.log('from here');
+        onResetFilter();
+      }}
+      onSubmit={async (values) => {
+        const payload = {
+          queryInput: values.queryInput.trim(),
+          scope: values.searchScope?.value,
+          type: values.searchType?.value,
+        };
+
+        await onApplyFilter(payload);
```

Comment:
Seems like it is not needed?

```suggestion
        onApplyFilter(payload);
```

<img width="823" alt="CleanShot 2023-09-12 at 13 01 30@2x" src="https://github.com/nimblehq/ic-nextjs-prabesh-dahal/assets/76901736/294e4645-fb8a-4906-81da-f69c96e0d2b6">